debian/changelog

```diff
@@ -1,9 +1,12 @@
-cloud-init (23.4-0ubuntu1~23.04.2) UNRELEASED; urgency=medium
+cloud-init (23.4.1-0ubuntu1~23.04.1) lunar; urgency=medium
 
   * d/p/retain-apt-pre-deb822.patch:
     - Disable apt source list generation with DEB822 style
+  * Upstream snapshot based on 23.4.1. (LP: #2045582).
```

Comment:
changeset, debian version and package build look good. What we miss is debian/changelog representation for the refreshed quilt patches, which we also missed in 23.4-0 -proposed upload previously too.
```diff
--- a/debian/changelog
+++ b/debian/changelog
@@ -2,6 +2,9 @@ cloud-init (23.4.1-0ubuntu1~23.04.1) lunar; urgency=medium
 
   * d/p/retain-apt-pre-deb822.patch:
     - Disable apt source list generation with DEB822 style
+  * refresh patches:
+    - d/p/status-do-not-remove-duplicated-data.patch
+  * d/changelog: amend 23.4-0 refresh patches entry
   * Upstream snapshot based on 23.4.1. (LP: #2045582).
     List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/23.4.1/ChangeLog
@@ -13,6 +16,8 @@ cloud-init (23.4-0ubuntu1~23.04.1) lunar; urgency=medium
   * d/p/status-do-not-remove-duplicated-data.patch:
     - Revert behavior downstream, leave duplicate data
   * d/control: add python3-apt as Recommends to read APT config from apt_pkg
+  * refresh patches:
+    - d/p/do-not-block-user-login.patch
   * Upstream snapshot based on 23.4. (LP: #2045582).
     List of changes from upstream can be found at
     https://raw.githubusercontent.com/canonical/cloud-init/23.4/ChangeLog
```